app/assets/javascripts/components/resources/resources.jsx

```diff
@@ -0,0 +1,71 @@
+
+import React from 'react';
+import { connect } from 'react-redux';
+import _ from 'lodash';
+
+import CourseLink from '../common/course_link.jsx';
+import { getWeeksArray } from '../../selectors';
+import Block from '../timeline/block';
+import TrainingModules from '../timeline/training_modules';
+import Handouts from './handouts';
+
+const Resources = ({ weeks, current_user, course }) => {
+  const trainingLibrarySlug = course.training_library_slug;
+  let instructorModulesLink;
+  if (current_user.isInstructor && Features.wikiEd) {
+    instructorModulesLink = <CourseLink to={'/training/instructors'} className="button pull-right">Instructor orientation modules</CourseLink>;
+  }
+  const blocks = _.flatten(weeks.map(week => week.blocks));
+  const modules = _.compact(_.flatten(blocks.map(block => block.training_modules)));
```

Comment:
Something like [the flow()](https://lodash.com/docs/4.17.11#flow) method might be useful here. Alternatively, just breaking this up into more variables might make it easier to read.

Reply:
How's this? https://github.com/WikiEducationFoundation/WikiEduDashboard/pull/2867/commits/8e1eb3291061ed624c6970d9c39ff939534ceefd

Reply:
Nice! I'm into this.  :)